src/mrinufft/operators/interfaces/bart.py

```diff
@@ -0,0 +1,285 @@
+        ]
+        try:
+            subp.run(cmd, check=True, capture_output=True)
+        except subp.CalledProcessError as exc:
+            msg = "Failed to run BART NUFFT\n"
+            msg += f"error code: {exc.returncode}\n"
+            msg += "cmd: " + " ".join(cmd) + "\n"
+            msg += f"stdout: {exc.output}\n"
+            msg += f"stderr: {exc.stderr}"
+            raise RuntimeError(msg) from exc
+
+        ksp_raw = _readcfl(self._ksp_file)
+        np.copyto(coeffs_data, ksp_raw)
+        return coeffs_data
+
+    def adj_op(self, coeffs_data, grid_data):
+        """Adjoint Operator."""
+        # Format grid data to cfl format, and write to file
+        # Run bart nufft with argument in subprocess
+
+        coeffs_ = coeffs_data.reshape(len(self.samples))
+        _writecfl(coeffs_[None, ..., None, None, None], self._ksp_file)
+
+        cmd = [
+            "bart",
+            "nufft",
+            "-d",
+            self.shape_str,
+            "-a" if "-i" not in self._adj_op_args else "",
+            *self._adj_op_args,
+            str(self._traj_file),
+            str(self._ksp_file),
+            str(self._grid_file),
+        ]
+        try:
+            subp.run(cmd, check=True, capture_output=True)
+        except subp.CalledProcessError as exc:
+            msg = "Failed to run BART NUFFT\n"
+            msg += f"error code: {exc.returncode}\n"
+            msg += "cmd: " + " ".join(cmd) + "\n"
+            msg += f"stdout: {exc.output}\n"
+            msg += f"stderr: {exc.stderr}"
+            raise RuntimeError(msg) from exc
+
+        grid_raw = _readcfl(self._grid_file)
+        np.copyto(grid_data, grid_raw)
+        return grid_data
+
+
+class MRIBartNUFFT(FourierOperatorCPU):
+    """BART implementation of MRI NUFFT transform."""
+
+    # TODO override Data consistency function: use toepliz
+
+    backend = "bart"
+    available = BART_AVAILABLE
+
+    def __init__(
+        self,
+        samples,
+        shape,
+        density=False,
+        n_coils=1,
+        n_batchs=1,
+        smaps=None,
+        squeeze_dims=True,
+        **kwargs,
+    ):
+        samples_ = proper_trajectory(samples, normalize="unit")
+        if density is True:
+            density = False
+            if getattr(kwargs, "extra_adj_op_args", None):
+                kwargs["extra_adj_op_args"] += ["-i"]
+            else:
+                kwargs["extra_adj_op_args"] = ["-i"]
+
+        super().__init__(
+            samples_,
+            shape,
+            density,
+            n_coils=n_coils,
+            n_batchs=n_batchs,
+            n_trans=1,
+            smaps=smaps,
+            squeeze_dims=squeeze_dims,
+        )
+
+        self.raw_op = RawBartNUFFT(samples_, shape, **kwargs)
+
+    @property
+    def norm_factor(self):
+        """Normalization factor of the operator."""
+        # return 1.0
+        return np.sqrt(2 ** len(self.shape))
+
+
+def _readcfl(cfl_file, hdr_file=None):
```

Comment:
It might as well be helpful to have this exposed (even the writer). We have collaborators who use CFL files for storing k-space data

Reply:
Sure, this could go in a proper io module (I guess we will tackle that in another PR) see #65